pymc_bart/tree.py

```diff
@@ -52,66 +96,88 @@ class Tree:
         "output",
     )
 
-    def __init__(self, tree_structure, idx_leaf_nodes, output):
+    def __init__(
+        self,
+        tree_structure: Dict[int, Node],
+        idx_leaf_nodes: Optional[List[int]],
+        output: Optional[npt.NDArray],
```

Comment:
These two parameters could be `None` as we have a method returning `Tree(tree, None, None)`  below. So I added some additional changes to check The none condition before calling methods on these attributes (e.g. calling `append` and `remove`). This is one of the benefits of mypy :)